src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> firstNode = new Node<>(null, value, null);
+            head = firstNode;
+            tail = firstNode;
+            size++;
+            return;
+        }
+        tail.next = new Node<>(tail, value, null);
+        tail = tail.next;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index >= 0 && size == 0 || index == size) {
```

Comment:
If size = 0, and we want to add element on index = 5 - is this the correct index to add element?

Reply:
This is checked in the findNodeByIndex method

Reply:
When we call method add, with idex = 5 and size = 0, value will be added without calling findNodeByIndex.

Reply:
![image](https://user-images.githubusercontent.com/57375800/140040646-b4b67a4d-a31d-420a-bc61-d2fe9221e2f5.png)